### ISSUE TITLE
due kanjis aren't refreshed on app start?

### ISSUE THREAD

**Issue by martinetd** [SYtor/Kanji-Dojo#114]
I've noticed this problem a few times but not sure if it's 100%: when starting kanji dojo in the morning I often have no kanji dues, then I fiddle around with the app a bit (either add a new set or review some readings I was putting off), and when that is done there are plenty of writing reviews due.

I haven't looked at the code for this one yet; this is on develop branch as of a few days ago (769d0bebba9a33b8a4ae947b4857d30438effe95) so perhaps a new bug if others haven't complained yet.

This is annoying enough so I'll probably have a look next weekend if nobody beat me to it

**Comment by syt0r** [comment]
You might be leaving the app open, it doesn't automatically refresh the list at the midnight for now

**Comment by martinetd** [comment]
Thanks for the reply!
Right, that's definitely possible, I'm not doing much else with the phone so it's quite likely the app didn't close.

I'll try to find time to add a timer or something.

(Note we might not want to make it midnight, but that's probably a separate topic -- most day-based review apps I know have a configurable time to make day change at 3am or something for folks working late)

**Comment by syt0r** [comment]
There is another feature request to adjust reset time #57 

Regarding this auto reset, I'm not sure if this is a good user experience to update data while in the app, suddenly making loading start while the user is in the middle of something. Does it work like this in other apps? It might be reasonable if a device is idling, but it sounds pretty tricky to track it

**Comment by martinetd** [comment]
Right, I agree we definitely shouldn't "reset" the app if it's actively open or in use!

I was thinking this could be done in the background when timer fires, but perhaps it'll be more natural to check if day changed whenever the app is displayed on home screen -- for example bank apps will prompt for passcode again whenever you switch focus and back, so there must be a way to detect this and give us an opportunity to recompute decks if they're outdated?
Or maybe check date whenever a deck is expanded, I think it's fine if home page reloads when you try to click on a deck if it's out of date as I'd definitely prefer getting up to date review lists than old ones...

**Comment by syt0r** [comment]
Hmm, it might be a good idea to track the date change event on the dashboard screen and show a snackbar message with refresh button, if it's not an automatic loading but a popup with a button then I think it should be alright
